File: application/src/main/java/de/htwberlin/schbuet/application/data/body/BodyProduct.java
package de.htwberlin.schbuet.application.data.body;

import lombok.Data;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotBlank;

@Data
public class BodyProduct {

    @NotBlank(message = "Name is mandatory")
    private String name;

    @NotBlank(message = "Description is mandatory")
    private String description;

    @NotBlank(message = "Category is mandatory")
    private String category;

    @NotBlank(message = "Item number is mandatory")
    private String itemNumber;

    private int priceInCents;

    @Min(1900)
    private int YearOfProduction;
}

File: frontend/src/app/models/body-product.ts
import {GeoAddress} from "./geo-address";

export class BodyProduct {
  name: string;
  description: string;
  category: string;
  itemNumber: string;
  address: GeoAddress;
  priceInCents: number;
  yearOfProduction: number;
  quantity: number;
  deliveryTimeInDays: number;

  constructor() {
    this.name = '';
    this.description = '';
    this.category = '';
    this.itemNumber = '';
    this.address = new GeoAddress();
    this.priceInCents = 0;
    this.yearOfProduction = 1900;
    this.quantity = 1;
    this.deliveryTimeInDays = 1;
  }
}
